### PR TITLE
Added Test for purchase, authorize, refund ,capture & void methods

### PR DESCRIPTION
**Pull request by gopalshimpi**
It is not finished yet, still working on it.
1. Altered library cams.ex 
2. Added Mock Tests for Purchase method.

Checklist:-
- [x] Add all the test cases
- [x] Test coverage should be 80% or more for cams.

### CODE REVIEW

File: test/gateways/cams_test.exs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  ]
+
+  @money 100
+  @bad_money "G"
+
+  describe "purchase" do
+
+    test "test_sucessful_purchase" do
+      with_mock HTTPoison,
+      [post: fn(_url, _body, _headers) -> MockResponse.successful_purchase_response end] do
```
Comment (arjun289) [inline test/gateways/cams_test.exs]: It's better to put a tab here to show it's continuation of the above line.

File: test/gateways/cams_test.exs [hunk 2]
```diff
@@ -0,0 +1,86 @@
+defmodule Gringotts.Gateways.CamsTest do
+
+  Code.require_file "../mocks/cams_mock.exs", __DIR__
+  use ExUnit.Case ,async: false
+	alias Gringotts.{
```
Comment (arjun289) [inline test/gateways/cams_test.exs]: use proper formatting here.

File: test/gateways/cams_test.exs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    company:  "Widgets Inc",
+    city:     "Ottawa",
+    state:    "ON",
+    zip:      "K1C2N6",
+    country:  "US",
+    phone:    "(555)555-5555",
+    fax:      "(555)555-6666"
+  }
+  @options  [
+    config: %{
```
Comment (arjun289) [inline test/gateways/cams_test.exs]: config: %{
 +    username: "testintegrationc",
 +    password: "password9"
 + },   
keep it this way

File: test/gateways/cams_test.exs (hunk 1 above)
Comment (gopalshimpi) [reply test/gateways/cams_test.exs]: Thanks of mentioning.

File: lib/gringotts/gateways/cams.ex [hunk 4]
```diff
@@ -1,36 +1,49 @@
...
 
-    * **Capture**   In  `capture/3`
+    Most `Gringotts` API calls accept an optional `Keyword` list `opts` to supply
+    optional arguments for transactions with the Cams gateway. The following keys
+    are supported:
+    
+    | Key                 | Remark | Status          |
+    | ----                | ---    | ----            |
+    | `billing_address`   |        | Not implemented |
+    | `address`           |     	 | Not implemented |
```
Comment (pkrawat1) [inline lib/gringotts/gateways/cams.ex]: @gopalshimpi the formatting is totally out of places. please make sure to review your PR in github also once you push your changes.

File: lib/gringotts/gateways/cams.ex [hunk 5]
```diff
@@ -58,10 +71,7 @@ defmodule Gringotts.Gateways.Cams do
           |> add_invoice(money, options)
```
Comment (pkrawat1) [inline lib/gringotts/gateways/cams.ex]: @gopalshimpi indentation is I guess 2 tabs with 4 spaces each. use 2 spaces and covert all tabs to spaces. again review your changes in the browser.

File: test/gateways/cams_test.exs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    order_id: 0001,
+    billing_address: @address,
+    description: "Store Purchase",
+  ]
+
+  @money 100
+  @bad_money "G"
+
+  describe "purchase" do
+
```
Comment (pkrawat1) [inline test/gateways/cams_test.exs]: @gopalshimpi this extra line is not required.

File: test/mocks/cams_mock.exs [hunk 7]
```diff
@@ -0,0 +1,30 @@
+defmodule Gringotts.Gateways.CamsMock do
```
Comment (pkrawat1) [inline test/mocks/cams_mock.exs]: @gopalshimpi the indentation is totally out of place in this whole use 2 spaces and verify once in github also.

>> Tip: when you are creating a new file in the starting change its setting to use spaces for indentation and that too 2 spaces.

File: lib/gringotts/gateways/cams.ex (hunk 5 above)
Comment (gopalshimpi) [reply lib/gringotts/gateways/cams.ex]: @pkrawat1 I have made necessary changes which pointed by you

Comment (jyotigautam) [reply lib/gringotts/gateways/cams.ex]: Remove address struct if it's unused.

File: lib/gringotts/gateways/cams.ex [hunk 8]
```diff
@@ -164,76 +186,105 @@ defmodule Gringotts.Gateways.Cams do
...
+
+    defp set_success(opts, %{"response_code" => response_code}) do
+      opts ++ [success: response_code == "100"]
+    end
 
-  defp extract_auth(post, response) do
-    response_body = URI.decode_query(response)
-    response_body["transactionid"]
   end
 end
```
Comment (jyotigautam) [inline lib/gringotts/gateways/cams.ex]: Leave a line at the end of the file as a general practice.

File: test/mocks/cams_mock.exs [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+    {"Content-Length", "155"}, {"Content-Type", "text/html; charset=UTF-8"}],
+    request_url: "https://secure.centralams.com/gw/api/transact.php",
+    status_code: 200}} 
+  end
+ 
+end
+
+
+
+
```
Comment (jyotigautam) [inline test/mocks/cams_mock.exs]: Indentation is not proper and so many blank lines have been added at the end of the file.

**Pull request merged**